lib/elixir/test/elixir/kernel/guard_test.exs

```diff
@@ -0,0 +1,111 @@
+Code.require_file "../test_helper.exs", __DIR__
+
+defmodule Kernel.GuardTest do
+  use ExUnit.Case, async: true
+
+  # Need to find a way to actually catch these.
+  #
+  # describe "Kernel.defguard usage" do
+  #   test "requires a proper name" do
+  #     assert_raise ArgumentError, fn ->
+  #       Kernel.defguard "asdf"
+  #     end
+  #   end
+  # end
+  #
+  # describe "Kernel.defguard compilation" do
+  #   test "fails on expressions not allowed in guard" do
```

Comment:
You need to use defmodule inside the test and define the guard inside that module.